src/python/WMComponent/JobCreator/CreateWorkArea.py

```diff
@@ -35,35 +29,20 @@ def createDirectories(dirList):
     Create the directory if everything is sane
 
     """
-
-
-    #This is gonna be tricky
-    cmdList = []
-    cmdArgs = ['mkdir']
-
-    while len(dirList) > 500:
-        cmdArgs.extend(dirList[:500])
-        cmdList.append(cmdArgs)
+    for sdirList in grouper(dirList, 500):
         cmdArgs = ['mkdir']
-        dirList = dirList[500:]
-    if len(dirList) > 0:
-        cmdArgs.extend(dirList)
-        cmdList.append(cmdArgs)
-
-    logging.info('Executing makedir commands')
-    for command in cmdList:
-        pipe = Popen(command, stdout = PIPE, stderr = PIPE, shell = False)
+        cmdArgs.extend(sdirList)
+        pipe = Popen(cmdArgs, stdout = PIPE, stderr = PIPE, shell = False)
```

Comment:
For future reference, there is a mkdir in python: https://docs.python.org/2/library/os.html#os.mkdir and for stuff where you don't need returns (not sure about here) there are much simpler ways: https://docs.python.org/2.6/library/subprocess.html#convenience-functions